lib/plugin.js

```diff
@@ -75,7 +76,12 @@ ManifestPlugin.prototype.apply = function(compiler) {
     // This allows output path to be reflected in the manifest.
     if (this.opts.basePath) {
       manifest = _.reduce(manifest, function(memo, value, key) {
-        memo[this.opts.basePath + key] = this.opts.basePath + value;
+        // only if useBasePathAndPublicPath is set, both basePath and publicPath are used
+        if (this.opts.useBasePathAndPublicPath && this.opts.publicPath) {
```

Comment:
Updated!

It has the same behavior now, but there are 3 if statements for the 3 situations.